platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapboxMap.java

```diff
@@ -981,7 +981,14 @@ public MarkerView addMarker(@NonNull BaseMarkerViewOptions markerOptions) {
         long id = mapView.addMarker(marker);
         marker.setId(id);
         annotations.put(id, marker);
-        markerViewManager.invalidateViewMarkersInVisibleRegion();
+
```

Comment:
@tobrun Instead of posting with a delay, which will probably work in most situations, but doesn't give any guarantees. How about using an `OnMapChangedListener` for the callback that listens for the appropriate signal?